src/vs/editor/common/viewLayout/viewLineRenderer.ts

```diff
@@ -513,9 +569,16 @@ function _applyRenderWhitespace(lineContent: string, len: number, continuesWithW
 	tmpIndent = tmpIndent % tabSize;
 
 	let wasInWhitespace = false;
+	let currentSelectionIndex = 0;
+	let currentSelection = selections && selections[currentSelectionIndex];
 	for (let charIndex = fauxIndentLength; charIndex < len; charIndex++) {
 		const chCode = lineContent.charCodeAt(charIndex);
 
+		if (currentSelection && charIndex > currentSelection.endOffset) {
```

Comment:
Here the check should be `charIndex >= currentSelection.endOffset`. Reverse-engineered steps to observe a bug:

- have some text ` * ` (`<space>`, `*`, `<space>`).
- create 3 multiple cursors before the first space, before the `*` and before the second space.
- press shift+right
- the whitespace is not painted in the second `<space>` due to this check.

This case could serve as a good unit test.

Actual:
![image](https://user-images.githubusercontent.com/5047891/61621965-2e087200-ac74-11e9-8e60-5f58c518b1af.png)

Expected:
![image](https://user-images.githubusercontent.com/5047891/61622085-63ad5b00-ac74-11e9-9a96-af8eae311743.png)

Reply:
Nice catch! I added a test for this